### PR TITLE
adjusted lowest brightness threshold for TS101

### LINKED ISSUE #2016: TS101 Screen Brightness Control
**Describe the bug**
 a weird issue with my TS101 display brightness when at 1 the screen appears to be off only at brightness level 2 then the screen lights up 

Brightness setting 2 - 5  work with no issues 

**To Reproduce**

 - Device: [e.g. TS101 ]
>  - Release: [v2.22.DED422AA4 ( 2024-12-07 ) & v2.22E.C308FE8C ( 05-11-23 )]
 - Power adapter being used: 5v PC USB 

Additional information 

 only at brightness level 2 then screen lights up.

( Brightness setting 2 - 5 work with no issues ) 

( is this a bug for the TS101 or have you already heard about it? )

### PR DESCRIPTION
**Pull request by discip**
should fix #2016

### CODE REVIEW

**Comment by discip** [comment]
@Gamerx007
Maybe you could give this a try? (https://github.com/Ralim/IronOS/actions/runs/12231216889/artifacts/2292295547)

**Comment by Gamerx007** [comment]
I will try updating my TS101 when I get back home today 

thanks again 😎👍

**Comment by Gamerx007** [comment]
Just got home tried to download but I am unable to find the file do you have another link 

Thanks :)

**Comment by discip** [comment]
@Gamerx007
Sorry here you go:
https://github.com/Ralim/IronOS/actions/runs/12233996481/artifacts/2293055357

**Comment by Gamerx007** [comment]
Just uploaded the test firmware to My TS101

the screen now works at level 1 brightness

the Screen is still a bit too bright in comparison to my TS100, My TS80, and My TS80P

hope this info helps 

any other questions feel free :) 

I can try to test for you and report back if you need I have now used my TS101 yet for soldering same goes for my TS80P.

when soldering I use my TS100 and My TS80 ;) 

Thanks again

**Review by Ralim — APPROVED** [review]

File: source/Core/BSP/Miniware/configuration.h [hunk 1]
```diff
@@ -63,8 +63,13 @@
  * OLED Brightness
  *
  */
-#define MIN_BRIGHTNESS     0   // Min OLED brightness selectable
-#define MAX_BRIGHTNESS     100 // Max OLED brightness selectable
+#if defined(MODEL_TS101)
+  #define MIN_BRIGHTNESS     1   // Min OLED brightness selectable
+  #define MAX_BRIGHTNESS     101 // Max OLED brightness selectable
```
Comment (ia) [inline source/Core/BSP/Miniware/configuration.h:68]: Hello. @Ralim & @discip, I have a technical question about this. The more I was thinking about this, the more I was getting very curious...

TL;DR: is it really safe & reliable to write `101` as brightness level directly to the _OLED hardware_?

- _OLED brightness_ is managed [here](https://github.com/Ralim/IronOS/blob/dev/source/Core/Src/settingsGUI.cpp#L872) by `displayBrightnessLevel()`
- then we jump into `OLED::setBrightness(...)` [here](https://github.com/Ralim/IronOS/blob/dev/source/Core/Drivers/OLED.cpp#L522)
- there we take a user "input" for brightness setting as [`contrast`](https://github.com/Ralim/IronOS/blob/dev/source/Core/Drivers/OLED.cpp#L524)
- and then, in short, as far as I understand, we fall through writing the setting value into _the OLED hardware_ through _software OLED driver_ using its register (which is allocated as a region of memory) using _I2C_ bus...

At the same time we have _"sanitizing"_ function to (re)set settings with validation which is [setSettingValue(...)](https://github.com/Ralim/IronOS/blob/dev/source/Core/Src/Settings.cpp#L179), where part of the settings matrix (min/max) is used. But! Setting _OLED brightness_ is not utilizing this "sanitization". Now, why it's important. _It depends on hardware_, so _some hardware_ _may be_ **fault tolerant** to wrong and/or fuzzing inputs but _**some may not**_. And something tells me that `100` as max brightness has been set originally based on datasheets of particular _OLED hardware_ used in the irons (haven't had time to dig so deep though yet).

Although I fully understand why minimum level for `TS101` should be `1`, is it OK to set maximum as `101` instead of original `100`?

As a safety measure, **one of two approaches** (_2a_ or _2b_) could be used for validation:

1. [set](https://github.com/ia/IronOSf/blob/6900fe81a181c02594bb1ce66928a07aed95db4c/source/Core/BSP/Miniware/configuration.h#L71) `MAX_BRIGHTNESS` back to `100` for _TS101_;
2. add validation (so even when it will be `1 + 25 + 25 + 25 + 25 = 101`, the actual setting for _the hardware_ still would be `100`):
  a. [directly](https://github.com/ia/IronOSf/blob/4da9015426743b2618d1991c1798bb2426d91f27/source/Core/Drivers/OLED.cpp#L523) inside of `OLED::setBrightness(...)`, i.e. `contrast = (MAX_BRIGHTNESS < contrast) ? MAX_BRIGHTNESS : contrast;`
  b. [indirectly](https://github.com/ia/IronOSf/blob/d1f60a7a966b691b76b11d969d971feb9c3b87a6/source/Core/Src/settingsGUI.cpp#L874-L876) inside of `displayBrightnessLevel()` using `setSettingValue()`, i.e. `setSettingValue(SettingsOptions::OLEDBrightness, value);` (the other calls of `OLED::setBrightness` either on start-up to set setting which is already there or via Bluetooth which is not related to _TS101_)

Should I make _pull-request_ with the check above or I shouldn't be worry much about this? Just really curious to hear any feedback on my thoughts (some parts describe very basic things but I did it only for my own better understanding).

**Review by ia — COMMENTED** [review]
Just curious to check some of my assumptions and to validate my own understanding of some parts of the code.

**Comment by discip** [comment]
@ia
TL;DR: With a value of **`101`** we are still safe, as the max value would be **`255`**. 👍
For more details please consult this post: https://github.com/Ralim/IronOS/issues/1078#issuecomment-929212314

So no action needed. 😊

**Comment by discip** [comment]
@ia
TL;DR: With a value of **`101`** we are still safe, as the max value would be **`255`**. 👍
For more details please consult these posts:
https://github.com/Ralim/IronOS/issues/1078#issuecomment-929212314
https://github.com/Ralim/IronOS/pull/1087#issuecomment-930581068

So no action needed. 😊

**Comment by ia** [comment]
@discip 

> TL;DR: With a value of **`101`** we are still safe, as the max value would be **`255`**. 👍 For more details please consult these posts:
> [#1078 (comment)](https://github.com/Ralim/IronOS/issues/1078#issuecomment-929212314)
> [#1087 (comment)](https://github.com/Ralim/IronOS/pull/1087#issuecomment-930581068)

Thank you a lot for digging out these links for me - very interesting and very educational. Much appreciated!

> So no action needed. 😊

Aha, got it.
